opencsp/common/lib/render/test/test_color.py

```diff
@@ -0,0 +1,122 @@
+        self.assertEqual(color.rgb_255()[1], 84)
+        self.assertEqual(color.rgb_255()[2], 126)
+
+    def test_to_hsv(self):
+        hsv = cl.Color(0.16470588, 0.32941176, 0.49411764, "The Answer", "DA").to_hsv()
+        self.assertAlmostEqual(hsv[0], 210 / 360.0, places=5)
+        self.assertAlmostEqual(hsv[1], 66.667 / 100.0, places=5)
+        self.assertAlmostEqual(hsv[2], 126 / 255.0, places=5)
+
+    def test_from_hsv(self):
+        color = cl.Color.from_hsv(210 / 360.0, 66.667 / 100.0, 126 / 255.0, "The Answer", "DA")
+        self.assertAlmostEqual(color.rgb()[0], 0.16470588, places=5)
+        self.assertAlmostEqual(color.rgb()[1], 0.32941176, places=5)
+        self.assertAlmostEqual(color.rgb()[2], 0.49411764, places=5)
+
+    def test_colormap_2(self):
+        color_map = cl.red().build_colormap(cl.blue())
+
+        # red
+        self.assertAlmostEqual(color_map(0)[0], 1.0, places=2)
+        self.assertAlmostEqual(color_map(0)[1], 0.0, places=2)
+        self.assertAlmostEqual(color_map(0)[2], 0.0, places=2)
+
+        self.assertAlmostEqual(color_map(0.5)[0], 0.5, places=2)
+        self.assertAlmostEqual(color_map(0.5)[1], 0.0, places=2)
+        self.assertAlmostEqual(color_map(0.5)[2], 0.5, places=2)
+
+        # blue
+        self.assertAlmostEqual(color_map(1.0)[0], 0.0, places=2)
+        self.assertAlmostEqual(color_map(1.0)[1], 0.0, places=2)
+        self.assertAlmostEqual(color_map(1.0)[2], 1.0, places=2)
+
+    def test_colormap_3(self):
+        color_map = cl.red().build_colormap(cl.green(), cl.blue())
+
+        # red
+        self.assertAlmostEqual(color_map(0)[0], 1.0, places=2)
+        self.assertAlmostEqual(color_map(0)[1], 0.0, places=2)
+        self.assertAlmostEqual(color_map(0)[2], 0.0, places=2)
+
+        self.assertAlmostEqual(color_map(0.25)[0], 0.5, places=2)
+        self.assertAlmostEqual(color_map(0.25)[1], 0.5, places=2)
+        self.assertAlmostEqual(color_map(0.25)[2], 0.0, places=2)
+
+        # green
+        self.assertAlmostEqual(color_map(0.5)[0], 0.0, places=2)
+        self.assertAlmostEqual(color_map(0.5)[1], 1.0, places=2)
+        self.assertAlmostEqual(color_map(0.5)[2], 0.0, places=2)
+
+        self.assertAlmostEqual(color_map(0.75)[0], 0.0, places=2)
+        self.assertAlmostEqual(color_map(0.75)[1], 0.5, places=2)
+        self.assertAlmostEqual(color_map(0.75)[2], 0.5, places=2)
+
+        # blue
+        self.assertAlmostEqual(color_map(1.0)[0], 0.0, places=2)
+        self.assertAlmostEqual(color_map(1.0)[1], 0.0, places=2)
+        self.assertAlmostEqual(color_map(1.0)[2], 1.0, places=2)
+
+    def test_matplotlibcolors_match(self):
+        mpl_colors = matplotlib.color_sequences['tab10']
+
+        for i, color in cl.plot_colorsi.items():
+            r1, g1, b1 = color.rgb()
+            r2, g2, b2 = mpl_colors[i]
+            self.assertAlmostEqual(r1, r2)
+            self.assertAlmostEqual(g1, g2)
+            self.assertAlmostEqual(b1, b2)
```

Comment:
Is this comparing the _plotColors class with the matplotlib tab10 colors? 
Why would plot_colors_names also not be tested in the unit tests? Is this since throughout this file it calls the colors by name? Such as cl.blue

Reply:
The purpose of this test isn't to verify that the color names match those from matplotlib, but rather to verify that the color values continue to match those in matplotlib into the future. I added the following comment to this test:

    The matplotlib tab10 colors are encoded in color.py. This test is here
    to catch any change in the matplotlib colors, if there ever is any.

Hopefully this answers your question?